### ISSUE TITLE
In R 3.3.1, state.x77 reg:linear fails

### ISSUE THREAD

**Issue by AndreMikulec** [dmlc/xgboost#1632]
Hi, I am trying to do the simplest linear regression

I am getting the answer

``` r
xgboost: Invalid input of data.
```

I do not see any bad data.
See below.

Steps to reproduce ...

The non-response data and response (vector) follows.

``` r

mytrain <- list()
mytrain[["data"]]   <- Matrix(state.x77[1:25,-1])
mytrain[["label"]]  <-        state.x77[1:25, 1]
```

I do not see any bad data types

``` r
> str(state.x77)
 num [1:50, 1:8] 3615 365 2212 2110 21198 ...
 - attr(*, "dimnames")=List of 2
  ..$ : chr [1:50] "Alabama" "Alaska" "Arizona" "Arkansas" ...
  ..$ : chr [1:8] "Population" "Income" "Illiteracy" "Life Exp" ...

> str(Matrix(state.x77))
Formal class 'dgeMatrix' [package "Matrix"] with 4 slots
  ..@ x       : num [1:400] 3615 365 2212 2110 21198 ...
  ..@ Dim     : int [1:2] 50 8
  ..@ Dimnames:List of 2
  .. ..$ : chr [1:50] "Alabama" "Alaska" "Arizona" "Arkansas" ...
  .. ..$ : chr [1:8] "Population" "Income" "Illiteracy" "Life Exp" ...
  ..@ factors : list()

> str(as.matrix(Matrix(state.x77)))
 num [1:50, 1:8] 3615 365 2212 2110 21198 ...
 - attr(*, "dimnames")=List of 2
  ..$ : chr [1:50] "Alabama" "Alaska" "Arizona" "Arkansas" ...
  ..$ : chr [1:8] "Population" "Income" "Illiteracy" "Life Exp" ...


> str(state.x77[1:25, 1])
 Named num [1:25] 3615 365 2212 2110 21198 ...
 - attr(*, "names")= chr [1:25] "Alabama" "Alaska" "Arizona" "Arkansas" ...

Named num [1:25] 3615 365 2212 2110 21198 ...
 - attr(*, "names")= chr [1:25] "Alabama" "Alaska" "Arizona" "Arkansas" ...

```

``` r

> Sys.Date()
[1] "2016-10-03"

> install.packages("xgboost")
# fit my model

mybst <- xgboost(data = mytrain$data, label = mytrain$label
  , objective = "reg:linear")

Error in xgb.get.DMatrix(data, label) : xgboost: Invalid input of data
In addition: Warning message:
In xgb.get.DMatrix(data, label) : xgboost: label will be ignored.
```

I tried nothing else.
Nothing else.
This is too simple.

Thanks.

``` r
> sessionInfo()
R version 3.3.1 (2016-06-21)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 10586)

locale:
[1] LC_COLLATE=English_United States.1252
[2] LC_CTYPE=English_United States.1252
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C
[5] LC_TIME=English_United States.1252

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] Matrix_1.2-7.1      randomForest_4.6-12 xgboost_0.4-4

loaded via a namespace (and not attached):
[1] magrittr_1.5     tools_3.3.1      stringi_1.1.1    grid_3.3.1
[5] data.table_1.9.6 stringr_1.1.0    chron_2.3-47     lattice_0.20-34
```

**Comment by khotilov** [comment]
Your `mytrain$data` is `dgeMatrix` which is not one of the acceptable inputs.

I admit that the error message could have been more specific, and there could have been handling of more of the `Matrix` package classes implemented. But so far please read the docs for the acceptable formats.

**Comment by AndreMikulec** [comment]
Where?

> But so far please read the docs for the acceptable formats.

I looked here.
[XGBoost R Tutorial](http://xgboost.readthedocs.io/en/latest/R-package/xgboostPresentation.html)

I spent 5 minutes all over google looking for a 'simple working example' of R language xgboost objective = "reg:linear"   I did not find one.  

I just added that information to the issue title. . . 
title+ = R language docs; reg:linear needs a simple working example
